main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1369,24 +1369,41 @@ object Lowering {
 
   /**
     * Returns a list of `GetChannel` expressions based on `symExps`.
+    *
+    * If the expression is either [[Expression.Var]] or a literal expression a [[Expression.GetChannel]] is **not** generated.
+    *
     */
   private def mkParWaits(symExps: List[(Symbol.VarSym, Expression)]): List[Expression] = {
     // Make wait expressions `<- ch, ..., <- chn`.
+    var first = true
```

Comment:
Trust the force, i mean, functional programming

Reply:
Yeah this was just meant as an experiment